src/Storages/IStorage.cpp

```diff
@@ -55,8 +55,8 @@ TableLockHolder IStorage::lockForShare(const String & query_id, const std::chron
 
     if (is_dropped)
     {
-        auto table_id = getStorageID();
-        throw Exception(ErrorCodes::TABLE_IS_DROPPED, "Table {}.{} is dropped", table_id.database_name, table_id.table_name);
+        // Table was dropped while acquiring the lock
+        result = nullptr;
```

Comment:
lockForShare is used in 50+ places. I doubt all of them have been adapted to now handle a nullptr and a non-existent table. Why not create a new function that doesn't throw instead? Something like `tryLockForShare()`